### DISCUSSION TITLE
Complete support block (no supports in a volume)

### DISCUSSION THREAD

**Discussion by foresterr** [OrcaSlicer/OrcaSlicer#5330] [Ideas]
After messing a bit with paint-on supports, and support blockers and enforcers, I realized that they all control where the supports will attach to (or, in other words, what they will support). But there is no way to control where the support will "grow out" from. E.g. excluding parts of build plate, or parts of the model itself. This is actually a somewhat frequent problem for functional parts (hinges, brackets etc) - let's say you have a (relatively small) vertical hole in your model, and somewhere above that hole, an overhang needing support. You of course mark the overhang as needing support, and then Orca will _absolutely insist_ on printing supports inside the hole. Where they will be a royal pain to remove, and also quite useless (the hole does not have enough top surface to contribute to the support in a meaningful way).

There is no intended way to block this. Support blockers can't control where the supports can't grow out from, only where they can't attach to. Negative parts are applied before support generation, so the supports will happily grow into spaces created by them. I have found a hack to achieve what I want - I can add a positive part, and set it to 0 walls, 0 infill, 0 top shell, 0 bottom shell. This results in empty space but Orca will consider it part of the model and not print _anything_ inside it, including supports. But it's still very much a hack - it cannot intersect with the real parts of the model, or it will cut holes through it (EDIT: it can actually, just need to change the order in object tree); also it's a total pain to manage (it just looks and acts like any other part of the model, and you can't visually distinguish it from the rest).

I'm wondering if anyone else sees it as a problem worth creating a feature request for. I think the hack described above could be rather easily adapted into a proper new modifier type, called "support excluder" or something (of course I don't know Orca's code so I might be grossly overoptimistic) - it would just have to be applied before the actual model parts, not after (so it does not cut holes in them; EDIT: I just realized I can reorder objects in the object tree and it's literally a matter of proper order), and have some proper UI controls (like showing it in a different color in the "Prepare" tab ;) )

**Comment by foresterr** [comment]
One additional observation why proper support for that type of modifier would be better than the hack described above - Orca might still try to generate supports anchored on top of the "hack" volume with no shell or infill. Unfortunately, they are unlikely to work if started in thin air (if it was possible we wouldn't need supports at all 😁)

**Comment by excieve** [comment]
This is still an issue. I'm trying to print a part with some hinges and this is killing me. Orca keeps generating a support that goes vertically inside the lower hinge to support the upper one. There's no reason to do this and removing this from inside the small hinge hole is near impossible. Support blockers on the lower hinge do nothing with it too.

**Comment by khill-fbmc** [comment]
I was hoping support blockers would help move the stump of the tree supports.

I want the supports to all "root" from _outside the cube_ and support each face's bridging as it arrives. Orca insists on "rooting" the support bases _inside the cube_.

I tried making a cylindrical support blocker in the center of the cube, and also a flat cube the same X and Y dimensions of my cube, but still.... the roots are inside the cube...

<img width="680" height="685" alt="image" src="https://github.com/user-attachments/assets/4ba6b5e8-2149-4cce-8a35-91a2ee2546bb" />
<img width="726" height="721" alt="image" src="https://github.com/user-attachments/assets/d5d641af-31a8-4e0d-9cf8-817b5caef7ce" />

**Comment by Evil-Crab** [comment]
Yeah. Here, for example, Orca generates support through the hole with no way to block it, even though supports on the model are enabled 
<img width="615" height="771" alt="image" src="https://github.com/user-attachments/assets/3ee5c7cb-c9a1-42ee-88d3-de1ac0c5b14c" />
<img width="472" height="550" alt="image" src="https://github.com/user-attachments/assets/433adcf3-c68f-46cd-a696-51f679fcd71d" />
